src/frontend/src/optimizer/plan_node/logical_join.rs

```diff
@@ -1110,11 +1146,35 @@ impl ToStream for LogicalJoin {
             .filter(|i| r2i.try_map(*i) == None)
             .map(|i| i + left_len);
 
+        // NOTE(st1page): add join keys in the pk_indices a work around before we really have stream
```

Comment:
Question: So you mean we don't have the actual "stream key" yet 🤔 Why? 

From my understanding the `logical_pk` in PlanNode is the stream key. Am I incorrect?

Reply:
Yes, now we just use `logical_pk` as the `stream_key` and add rewrite the plan tree from bottom to maintain every plan node has `logical_pk` in the locical phase.
But in future we will move the rewriting phase on the stream plan tree to maintain every plan node has `stream_key`. so that the stream_key can refer to `distribution key`, `append_only` and other stream properties